compiler/test/compilable/test23617.d

```diff
@@ -0,0 +1,23 @@
+// https://issues.dlang.org/show_bug.cgi?id=23617
+
+struct S
+{
+    void foo() {}
+}
+
+struct Wrapper
+{
+    size_t currentIndex;
+    S[] arrayOfS;
+
+    auto opDispatch(string name, T ...)(T t)
+    {
+        return __traits(child, arrayOfS[this.currentIndex], __traits(getMember, S, name))(t);
```

Comment:
it should also work without `this` on the `this.currentIndex` so might be worth adding that to the test

```
        __traits(child, arrayOfS[currentIndex], __traits(getMember, S, name))(t); // should work without this
        return __traits(child, arrayOfS[this.currentIndex], __traits(getMember, S, name))(t); // and with
```